rinja_parser/src/expr.rs

```diff
@@ -199,27 +198,27 @@ impl<'a> Expr<'a> {
     expr_prec_layer!(addsub, muldivmod, "+", "-");
     expr_prec_layer!(muldivmod, filtered, "*", "/", "%");
 
-    fn filtered(i: &'a str, level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
-        let (_, level) = level.nest(i)?;
-        let start = i;
-        let (i, (obj, filters)) =
-            tuple((|i| Self::prefix(i, level), many0(|i| filter(i, level))))(i)?;
-
-        let mut res = obj;
-        for (fname, args) in filters {
-            res = WithSpan::new(
-                Self::Filter(Filter {
-                    name: fname,
-                    arguments: {
-                        let mut args = args.unwrap_or_default();
-                        args.insert(0, res);
-                        args
-                    },
-                }),
-                start,
-            );
+    fn filtered(i: &'a str, mut level: Level) -> ParseResult<'a, WithSpan<'a, Self>> {
+        #[allow(clippy::type_complexity)]
+        fn filter<'a>(
+            i: &'a str,
+            level: &mut Level,
+        ) -> ParseResult<'a, (&'a str, Option<Vec<WithSpan<'a, Expr<'a>>>>)> {
+            let (i, _) = char('|')(i)?;
+            *level = level.nest(i)?.1;
+            pair(ws(identifier), opt(|i| Expr::arguments(i, *level, false)))(i)
         }
```

Comment:
This nested function is the same as [rinja_parser/src/lib.rs](https://github.com/rinja-rs/rinja/pull/20/files#diff-d7d4a4880df4695f086349de84bd46afd2546635da8c8c7480abe26933c0a82f), isn't it? I think you can delete it, then.

Reply:
Indeed! Removed it and used the one from `src/lib.rs` instead.